src/Unsafe/Linear.hs

```diff
@@ -32,3 +32,7 @@ toLinear2
      (a :: TYPE r1) (b :: TYPE r2) (c :: TYPE r3).
      (a -> b -> c) ->. (a ->. b ->. c)
 toLinear2 = coerce
+
+-- | Like 'toLinear' but for three-argument functions
+toLinear3 :: (a -> b -> c -> d) ->. (a ->. b ->. c ->. d)
+toLinear3 f = toLinear2 (\a b -> toLinear (f a b))
```

Comment:
```suggestion
toLinear3 = coerce
```

No need to be overly conservative here. Everything is unsafe in this module anyway.